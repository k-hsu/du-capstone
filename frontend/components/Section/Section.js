import styled from "@emotion/styled";
import { spacing } from "../../theme";

const StyledSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1 1 auto;
  padding: ${spacing["3"]} ${spacing["1"]};
  background: #dfdfdf;
  margin: 0;
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  max-width: 1240px;
  margin: 0;
  gap: ${spacing["1"]};
`;

const Section = ({ children }) => {
  return (
    <StyledSection>
      <Container>{children}</Container>
    </StyledSection>
  );
};

export default Section;
