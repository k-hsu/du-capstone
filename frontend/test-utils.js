import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
