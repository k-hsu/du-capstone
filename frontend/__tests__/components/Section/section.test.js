import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Section from "../../../components/Section/Section";

describe("Section", () => {
  it("should render section component", () => {
    render(<Section>part me</Section>);
    expect(screen.getByText("part me")).toBeInTheDocument();
  });
  it("should render section component styles", () => {
    const component = renderer.create(<Section>part me</Section>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
