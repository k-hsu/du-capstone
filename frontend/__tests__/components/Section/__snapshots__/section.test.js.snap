// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render section component styles 1`] = `
<section
  className="css-3qpy7j"
>
  <div
    className="css-xqmc9"
  >
    part me
  </div>
</section>
`;
